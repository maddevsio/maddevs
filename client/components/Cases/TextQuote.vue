<template>
  <blockquote class="case_blockquote">
    <slot></slot>
  </blockquote>
</template>

<script>
export default {
  name: 'TextQuote',
}
</script>

<style lang="scss" scoped>
@import '../../assets/styles/cases/_mixins';

.case_blockquote {
  display: flex;
  position: relative;
  padding-left: 62px;
  @include default_text($text-color--black-cases, 30px, 166%, -0.02em, normal);

  &:before {
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    width: 50px;
    height: 50px;
    @include red-quote;
  }

  @media screen and (max-width: 768px) {
    padding-left: 37px;
    @include default_text($text-color--black-cases, 24px, 150%, -0.02em, normal);

    &:before {
      width: 32px;
      height: 33px;
      top: 4px;
    }
  }
}
</style>
