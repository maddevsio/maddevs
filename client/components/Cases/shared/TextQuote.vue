<template>
  <blockquote class="case_blockquote">
    <img
      :data-src="require(`@/assets/img/Studies/svg/qmark.svg`)"
      alt="blockquote"
      class="case_blockquote-mark img_lazy"
    >
    <slot />
  </blockquote>
</template>

<script>
export default {
  name: 'TextQuote',
}
</script>

<style lang="scss" scoped>
@import '../../../assets/styles/cases/_mixins';

.case_blockquote {
  display: flex;
  position: relative;
  padding-left: 62px;
  @include default_text($text-color--black-oil, 30px, 166%, -0.02em, normal);

  &-mark {
    content: '';
    width: 50px;
    height: 50px;
    position: absolute;
    top: -5px;
    left: 0;
    display: block;
  }

  @media screen and (max-width: 768px) {
    padding-left: 37px;
    @include default_text($text-color--black-oil, 24px, 150%, -0.02em, normal);

    &-mark {
      width: 32px;
      height: 33px;
      top: 4px;
    }
  }
}
</style>
