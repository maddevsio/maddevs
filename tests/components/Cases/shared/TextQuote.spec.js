import TextQuote from '@/components/Cases/shared/TextQuote'
import { render } from '@testing-library/vue'

const slots = {
  default: 'Main Content',
}

describe('TextQuote component', () => {
  it('should render correctly', () => {
    const { container } = render(TextQuote, {
      slots,
    })

    expect(container).toMatchSnapshot()
  })
})
