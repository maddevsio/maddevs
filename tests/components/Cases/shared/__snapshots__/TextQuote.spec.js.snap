// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextQuote component should render correctly 1`] = `
<div>
  <blockquote
    class="case_blockquote"
  >
    <img
      alt="blockquote"
      class="case_blockquote-mark img_lazy"
      data-src="[object Object]"
    />
     Main Content
  </blockquote>
</div>
`;
